scripts/exglobal_prep_ocean_obs.py

```diff
@@ -24,30 +25,51 @@
 OCNOBS2IODAEXEC = os.getenv('OCNOBS2IODAEXEC')
 
 OBS_YAML = os.getenv('OBS_YAML')
+# this will fail with FileNotFoundError if all yaml files in OBS_YAML are not
+# present in OBS_YAML_DIR
 obsConfig = YAMLFile(OBS_YAML)
 
 OBSPROC_YAML = os.getenv('OBSPROC_YAML')
 obsprocConfig = YAMLFile(OBSPROC_YAML)
 
 # TODO (AFE): needs more error handling (missing sources, missing files)
-for observer in obsConfig['observers']:
+try:
+    # For each of the observation sources (observers) specificed in the OBS_YAML...
+    for observer in obsConfig['observers']:
 
-    obsSpaceName = observer['obs space']['name']
-    print(f"obsSpaceName: {obsSpaceName}")
+        try:
+            obsSpaceName = observer['obs space']['name']
+        except KeyError:
+            print(f"observer: {observer}")
+            print("WARNING: Ill-formed observer yaml file, skipping")
+            continue  # to next observer
 
-    for observation in obsprocConfig['observations']:
+        print(f"obsSpaceName: {obsSpaceName}")
+# ...look through the observations in OBSPROC_YAML...
+        for observation in obsprocConfig['observations']:
```

Comment:
Isn't there a better way than looping through a second list and finding a match? Like a key, value pair dict?